zerver/lib/send_email.py

```diff
@@ -588,3 +590,12 @@ def send_custom_email(
 
         if options["dry_run"]:
             break
+
+
+def log_email_config_errors() -> None:
+    """
+    The purpose of this function is to log (potential) config errors,
+    but without raising an exception.
+    """
+    if settings.EMAIL_HOST_USER and not settings.EMAIL_HOST_PASSWORD:
+        logger.error("EMAIL_HOST_USER is set but not EMAIL_HOST_PASSWORD")
```

Comment:
I was very undecided about whether we should raise an exception here as well or just log an error and continue.

Reply:
I can see arguments either way.  An exception might be better, but I don't think it matters -- you'll get an exception in a moment that authentication failed, with this just above that.